packages/api/src/services/emp-state.ts

```diff
@@ -145,6 +148,21 @@ export default (config: Config, appState: Dependencies) => {
       }
       // add any new sponsors
       await emps.active.addSponsors(address, eventState.sponsors || []);
+
+      // set created timestamp if needed
+      await updateCreatedTimestamp(address, emps.active);
+    }
+  }
+
+  async function updateCreatedTimestamp(address: string, table: uma.tables.emps.JsMap) {
+    if (registeredEmpsMetadata.has(address)) {
+      const emp = await table.get(address);
+      const blockMetadata = registeredEmpsMetadata.get(address);
+
+      if (!emp.createdTimestamp && blockMetadata) {
```

Comment:
i dont think you need to check for blockMetadata since you do this first `registeredEmpsMetadata.has(address)`

Reply:
I know, I did it to get rid of the possible `undefined` result returned by `.get()`. `provider.getBlock` function complained about it